### PR TITLE
Improve k8 agent integration

### PR DESCRIPTION
**Pull request by anthony-grasso-datastax**
* Moved calls to Astra DevOps API out of credentials and into client
  * Created functions to create and delete a token in Astra
* Changed the way token life cycle is managed
  * Removed `renewal_time` from config path and `lease_time` from credentials path
  * Added `ttl` and `max_ttl` to role where they are used during token renewal as part of the response
* Changed how token data is stored locally
  * Calculate a unique token hash based on OrgId, RoleName, and LogicalName
  * Use unique token hash to save, retrieve, and delete token from local storage
* Added sidecar mode
  * Added `caller_mode` switch to config path to allow sidecar mode to be enabled
  * In sidecar mode read and update are identical, and create a new token when called
  * In sidecar mode the token's clientId is used to save the token in local storage
  * In sidecar mode `logical_name` is ignored when calling the org/token path

Todo:
 - [x] Update tests
 - [x] Remove debug code
 - [x] Update logging
 - [x] Update documentation/README
 - [x] Bump version
 - [x] Squash commits and rebase on `main` branch

### CODE REVIEW

File: astra_client.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	"io"
+	"net/http"
+	"strings"
+
+	dsAstraClient "github.com/datastax/astra-client-go/v2/astra"
+)
+
+const (
+	secretsPath   = "/v2/clientIdSecrets"
+	pluginversion = "Vault-Plugin v1.0.0"
```
Comment (bansri29) [inline astra_client.go:14]: need to update the version here

File: astra_token.go [hunk 2]
```diff
@@ -10,15 +16,15 @@ const (
...
-	Token        string            `json:"token"`
-	GeneratedOn  string            `json:"generatedOn"`
-	Metadata     map[string]string `json:"metadata"`
-	LogicalName  string            `json:"logicalName"`
-	RoleNickname string            `json:"roleNickname"`
+	ClientID    string            `json:"clientId"`
+	Secret      string            `json:"secret"`
+	OrgID       string            `json:"orgId"`
+	RoleName    string            `json:"roleName"`
+	Roles       []string          `json:"roles"`
```
Comment (bansri29) [inline astra_token.go:23]: Is field 'Roles' ever used? If not we can get rid of it.

File: astra_client.go (hunk 1 above)
Comment (bansri29) [inline astra_client.go:14]: need to change the version here

**Review by vaishakhbn — APPROVED** [review]
This looks great! Thank you Anthony.

**Pull request merged**